forms/gridfield/GridFieldExportButton.php

```diff
@@ -115,12 +115,14 @@ protected function getExportColumnsForGridField(GridField $gridField) {
 	 * Generate export fields for CSV.
 	 *
 	 * @param GridField $gridField
-	 * @return array
+	 * @return string
 	 */
 	public function generateExportFileData($gridField) {
 		$separator = $this->csvSeparator;
 		$csvColumns = $this->getExportColumnsForGridField($gridField);
-		$fileData = '';
+
+		$stream = fopen('php://memory', 'w');
```

Comment:
I think the issue you had is that headers_sent() was getting flagged to true, even though you weren't really sending any content from this block. Maybe double check your output when downloading locally?

If you are still having issues, maybe just try writing it to a temp file.